sunpy/data/data_manager/tests/test_manager.py

```diff
@@ -0,0 +1,137 @@
+from pathlib import Path
+
+import pytest
+
+from sunpy.data.data_manager.tests.mocks import write_to_test_file, MOCK_HASH
+from sunpy.util.exceptions import SunpyUserWarning
+
+
+def test_basic(storage, downloader, data_function):
+    data_function()
+
+    assert downloader.times_called == 1
+    assert len(storage._store) == 1
+    assert Path(storage._store[0]['file_path']).name == ('test_file')
+
+
+def test_cache(manager, storage, downloader, data_function):
+    """
+    Test calling function multiple times does not redownload.
+    """
+    data_function()
+    data_function()
+
+    assert downloader.times_called == 1
+    assert len(storage._store) == 1
+    assert Path(storage._store[0]['file_path']).name == ('test_file')
+
+
+def test_file_tampered(manager, storage, downloader, data_function):
+    """
+    Test calling function multiple times does not redownload.
+    """
+    data_function()
+    write_to_test_file(manager._tempdir + '/test_file', 'b')
+    with pytest.warns(SunpyUserWarning):
+        data_function()
+
+    assert downloader.times_called == 2
+    assert len(storage._store) == 1
+    assert Path(storage._store[0]['file_path']).name == ('test_file')
+
+
+def test_wrong_hash_provided(manager):
+    @manager.require('test_file', ['url1'], 'wrong_hash')
+    def test_foo():
+        pass
+
+    with pytest.raises(RuntimeError):
+        test_foo()
+
+def test_skip_all(manager, storage, downloader, data_function):
+    """
+    Test skip_hash_check redownloads data.
+    """
+    data_function()
+    with manager.skip_hash_check():
+        data_function()
+
+    assert downloader.times_called == 2
+    assert len(storage._store) == 1
+    assert Path(storage._store[0]['file_path']).name == ('test_file')
+
+def test_replace_file(manager, storage, downloader, data_function):
+    """
+    Test the replace_file functionality.
+    """
+
+    def default_tester(manager):
+        """
+        Function to test whether the file name is test_file.
+        """
+        assert manager.get('test_file').name == ('test_file')
+
+    def replace_file_tester(manager):
+        """
+        Function to test whether the file is /tmp/another_file.
+        """
+        assert manager.get('test_file') == Path('/tmp/another_fiel')
```

Comment:
'/tmp/another_fiel' -> '/tmp/another_file' ? The 'fiel' misspelling occurs several other times in this test